Sprint 1

AleksandrMenshchikov opened the pull request.

AleksandrMenshchikov committed:
modified:   .gitignore
	new file:   .parcelrc
	new file:   .postcssrc
	new file:   app.js
	new file:   assets/fonts/Inter-Medium.woff
	new file:   assets/fonts/Inter-Medium.woff2
	new file:   assets/fonts/Inter-SemiBold.woff
	new file:   assets/fonts/Inter-SemiBold.woff2
	new file:   assets/fonts/fonts.css
	new file:   handlebars.config.js
	new file:   package-lock.json
	new file:   package.json
	new file:   src/pages/index.css
	new file:   src/pages/index.hbs
	new file:   src/pages/login/index.css
	new file:   src/pages/login/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/index.hbs
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	new file:   src/pages/signup/index.css
	new file:   src/pages/signup/index.hbs

AleksandrMenshchikov committed:
new file:   src/pages/404/index.css
	new file:   src/pages/404/index.hbs
	new file:   src/pages/500/index.css
	new file:   src/pages/500/index.hbs
	new file:   src/pages/chat/index.css
	new file:   src/pages/chat/index.hbs
	modified:   src/pages/index.hbs

AleksandrMenshchikov committed:
new file:   assets/search.svg
	modified:   src/pages/404/index.css
	modified:   src/pages/500/index.css
	modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs
	modified:   src/pages/index.css
	modified:   src/pages/index.hbs
	modified:   src/pages/login/index.css
	modified:   src/pages/signup/index.css

AleksandrMenshchikov committed:
modified:   handlebars.config.js
	modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs
	new file:   src/partials/chat/chat.hbs

AleksandrMenshchikov committed:
modified:   package.json
	modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.css

AleksandrMenshchikov committed:
new file:   assets/arrowLeft.svg
	new file:   assets/arrowRight.svg
	new file:   assets/avatar.svg
	new file:   assets/file.svg
	new file:   assets/foto.svg
	new file:   assets/location.svg
	modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs

AleksandrMenshchikov committed:
modified:   src/pages/chat/index.hbs

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   README.md

AleksandrMenshchikov committed:
modified:   README.md

mikebars1995 commented on package.json:
```diff
@@ -0,0 +1,35 @@
+  "dependencies": {
+    "express": "^4.18.2"
+  }
+}
```
**Нужно исправить**



По семантике нужно помещать весь главный контент страницы в тег `main`.

AleksandrMenshchikov replied:
Исправил

mikebars1995 commented on src/pages/404/index.css:
```diff
@@ -0,0 +1,50 @@
+  margin: 50px auto 0;
+  font-weight: 500;
+  font-size: 1.2rem;
+  color: #3369f3;
```
**Нужно исправить**



все **цвета** лучше вынести в `css`-переменные и использовать их, чтобы можно было мгновенно понимать, что это за цвет, и можно было бы подкорректировать его в одном месте  и не искать дублирования во всем коде

AleksandrMenshchikov replied:
Исправил

mikebars1995 commented on package.json:
```diff
@@ -0,0 +1,35 @@
+  },
+  "homepage": "https://github.com/AleksandrMenshchikov/messenger#readme",
+  "devDependencies": {
+    "@anikin/parcel-transformer-handlebars": "^1.0.9",
```
**Можно лучше**

Лучше задавать **фиксированные версии** зависимостям. Если нечаянно удалить `package-lock.json` и попытаться повторно установить зависимости с помощью команды `npm install`, то **npm** установит последние версии зависимостей, что может привести к ошибкам.

AleksandrMenshchikov replied:
Исправил

mikebars1995 commented on src/pages/404/index.hbs:
```diff
@@ -0,0 +1,19 @@
+        </div>
+    </div>
+  </body>
+</html>
```
**Нужно исправить**



В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`

AleksandrMenshchikov replied:
Исправил

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная

Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Хорошая структура папок и файлов
* Отлично, что не забываете про `alt` в тегах `img`





Что нужно исправить:


* По семантике нужно помещать весь главный контент страницы в тег `main`.
* все **цвета** лучше вынести в `css`-переменные и использовать их, чтобы можно было мгновенно понимать, что это за цвет, и можно было бы подкорректировать его в одном месте  и не искать дублирования во всем коде
* В проекте в некоторых файлах есть ошибка `EOF`. Для `git` важно наличие **пустой строки** в конце файла. Подробнее тут: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log Можно в настройки IDE добавить `insert_final_newline = true`


Что можно улучшить:

* У `Handlebars` есть возможность вставлять куски одинаковой верстки. Не нужно ее дублировать из шаблона в шаблон. Для этого нужно создать папку `layouts` и в ней создать файл, например: `base.hbs` с повторяющейся версткой. И в шаблонах эту верстку можно вставлять вот так: `{{!< base}}`. И Вы можете кучу таких файлов создать там и вставлять в верстку туда, где нужно. А есть еще `partials` (еще мельче блоки можно вставлять). Подробнее https://habr.com/ru/post/273581/ 
* Лучше задавать **фиксированные версии** зависимостям. Если нечаянно удалить `package-lock.json` и попытаться повторно установить зависимости с помощью команды `npm install`, то **npm** установит последние версии зависимостей, что может привести к ошибкам.




Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

AleksandrMenshchikov committed:
modified:   src/pages/404/index.css
	modified:   src/pages/404/index.hbs
	modified:   src/pages/500/index.css
	modified:   src/pages/500/index.hbs
	modified:   src/pages/chat/index.css
	modified:   src/pages/chat/index.hbs
	modified:   src/pages/index.hbs
	modified:   src/pages/login/index.css
	modified:   src/pages/login/index.hbs
	modified:   src/pages/signup/index.css
	modified:   src/pages/signup/index.hbs
	modified:   src/partials/chat/chat.hbs

AleksandrMenshchikov committed:
modified:   package.json
	new file:   src/layouts/base.hbs
	modified:   src/pages/404/index.hbs
	modified:   src/pages/500/index.hbs
	modified:   src/pages/chat/index.hbs
	modified:   src/pages/login/index.hbs
	modified:   src/pages/signup/index.hbs

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.

AleksandrMenshchikov committed:
Merge branch 'main' into sprint_1
